### ISSUE TITLE
IPv6 filtering does not work on Oreo 8.1 + Magisk

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#1679]
Source report:
https://forum.xda-developers.com/showpost.php?p=75076426&postcount=2958

**Comment by carlylemiii** [comment]
I can test this on my Pixel 2 XL with Magisk 15.2, Android 8.1, if need be, once a fix is developed. Just let me know :)

**Comment by romracer** [comment]
I was the reporter on XDA.  So I understand, what's the crazy workaround deployed for Oreo (8.0 I guess) and Magisk?  Is it the create_transparent_proxy binary stuff?

I can provide the Magisk module to patch SELinux that I'm using to get this working for now, but I don't have either of the Wi-Fi Calling providers so someone else would have to test that portion.

**Comment by ameshkov** [comment]
The workaround is inside of the `create_transparent_proxy` binary, smth like this:

```
    // Some SU programs use own security context which may have not enough rights for creating/transferring sockets
    // 'init' context has enough rights and used by SuperSU.
    // No-op on Android 4.x
    if (arg_reset_con) {
        char *con;
        if (getcon(&con) != -1) {
            // Don't update context for LineageOS embedded SU
            if (strcmp(con, "u:r:sudaemon:s0") != 0) {
                if (setcon("u:r:init:s0") == -1) {
                    fprintf(stderr, "Warning: can't set SELinux security context for socket sender: %s\n", strerror(errno));
                }
            }
            free(con);
        }
    }
```

**Comment by ameshkov** [comment]
Guys, could you please check this test build?
[adguard_ipv6_fix.apk.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1646813/adguard_ipv6_fix.apk.zip)

**Comment by carlylemiii** [comment]
I get a "There was a problem parsing the package" when I rename it to end in .apk. :(

**Comment by carlylemiii** [comment]
And I've tried downloading twice and renaming using two different apps.

**Comment by ameshkov** [comment]
Do not rename, unzip it:)

**Comment by carlylemiii** [comment]
I think it works again, since my WiFi Calling does not, heh. I also did a quick check of apkmirror.com, where I've immediately noticed the problem before otherwise.

https://www.dropbox.com/s/0ps0uruy1fkk7pe/adguard.log?dl=0

**Comment by ameshkov** [comment]
What should I look for in the log?
A screenshot maybe?

**Comment by carlylemiii** [comment]
Sorry, I thought the log includes an error if ipv6 setup doesn't happen appropriately? It was included as an extra if you wanted to double check

**Comment by ameshkov** [comment]
Oh, sorry, I thought something IS wrong with apkmirror:)

The log is okay, IPv6 filtering seems to work.

What about disabling AG for `Android OS` now? Does it help with Wi-Fi calling?

**Comment by carlylemiii** [comment]
Seems promising (as in, it works). I'll use proxy mode the rest of the day, to be sure. Now, to figure out how to make the icon hidden from my status bar again (I'm pretty sure I recall there being some sort of work around for that 8.1 change).

**Comment by carlylemiii** [comment]
No, the beta + Android OS filtering disabled still kills WiFi Calling.

**Comment by ameshkov** [comment]
@carlylemiii just in case, could you plz also disable AG for `com.google.android.dialer` (the Phone app)

**Comment by ameshkov** [comment]
And Google Play services:)

**Comment by ameshkov** [comment]
Okay, I've got one more idea, we need some time to prepare a test build.

**Comment by ameshkov** [comment]
[adguard_ipv6_fix2.apk.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1647386/adguard_ipv6_fix2.apk.zip)
[Uploading adguard_ipv6_fix2.apk.zip…]()

**Comment by carlylemiii** [comment]
Should I have any packages disabled, or use as-is?

**Comment by ameshkov** [comment]
Meh:( No idea so far:(

**Comment by ameshkov** [comment]
@carlylemiii can you use terminal with root access? I've just realized that it'd be much faster to troubleshoot by executing a couple of different commands one by one. Thus we'll be able to figure what exact instruction breaks it.

**Comment by carlylemiii** [comment]
Sure. Just give me maybe slightly more detail than you think I would need. :) (Can I use this, or do I need to do it through adb on a computer?)

https://play.google.com/store/apps/details?id=jackpal.androidterm&hl=en

**Comment by ameshkov** [comment]
As you wish. I prefer `adb` as it's easier to type on a normal keyboard, but that's just me:)

So, here are the commands.

### 0. Clean up
First of all, these 6 commands will remove all the custom rules. Run it before and after testing to clean everything up.

```
/system/bin/ip6tables -D OUTPUT -p udp --dport 80 -j DROP
/system/bin/ip6tables -D OUTPUT -p udp --dport 443 -j DROP
/system/bin/ip6tables -t mangle -F OUTPUT
/system/bin/ip6tables -t mangle -F PREROUTING
/system/bin/ip -6 rule del prio 8000 from all fwmark 26 lookup 800
/system/bin/ip -6 route flush table 800
```

Now to the tests. Run the commands below one by one and check when Wi-Fi calling stops working.

### 1. Blocking QUIC
```
/system/bin/ip6tables -A OUTPUT -p udp --dport 80 -j DROP
/system/bin/ip6tables -A OUTPUT -p udp --dport 443 -j DROP
```

### 2. Adding new route
```
/system/bin/ip -6 rule add prio 8000 from all fwmark 26 lookup 800
/system/bin/ip -6 route add local default dev lo table 800
```

### 3. Protecting existing connections
```
/system/bin/ip6tables -t mangle -A PREROUTING -i lo -p tcp -m socket -j RETURN
```

### 4. IPv6 redirect
These commands will redirect traffic to ports 80 and 443 to a non-existent proxy on port 1102. As there's no proxy on that port, it will effectively break IPv6, but anyway, it should have no effect on Wi-Fi calling in theory.

```
/system/bin/ip6tables -t mangle -A OUTPUT -p tcp --dport 80 -j MARK --set-mark 26
/system/bin/ip6tables -t mangle -A PREROUTING -i lo -p tcp --dport 80 -m mark --mark 26 -j TPROXY --on-port 1102 --on-ip ::1 --tproxy-mark 0x0
/system/bin/ip6tables -t mangle -A OUTPUT -p tcp --dport 443 -j MARK --set-mark 26
/system/bin/ip6tables -t mangle -A PREROUTING -i lo -p tcp --dport 443 -m mark --mark 26 -j TPROXY --on-port 1102 --on-ip ::1 --tproxy-mark 0x0
```

That's it, now you can run the clean up commands again.

**Comment by carlylemiii** [comment]
Am I starting this off right? 

<img width="391" alt="2018-01-21 14_49_46-command prompt - adb shell" src="https://user-images.githubusercontent.com/29067983/35198851-708133e2-feba-11e7-9cd3-2e28c93aa4c0.png">

Also is AdGuard supposed to be enabled or disabled, and if enabled, I assume proxy mode?

P.S. ADB is fine and works better for me for that reason as well. I just wasn't positive on terminology and if what you were asking me to do was done via ADB. :)

**Comment by carlylemiii** [comment]
Should AdGuard be enabled or disabled while doing all of this? (If enabled, proxy mode?)

**Comment by ameshkov** [comment]
> Should AdGuard be enabled or disabled while doing all of this? (If enabled, proxy mode?)

Disabled, we don't need AG at all to conduct this test

**Comment by carlylemiii** [comment]
I didn't even get to an official step!

`/system/bin/ip6tables -t mangle -F PREROUTING`

kills it.

Also,

`/system/bin/ip -6 rule del prio 8000 from all fwmark 26 lookup 800`

gives the error

> RTNETLINK answers: No such file or directory

(Maybe doesn't matter at this exact moment, but noting it since you said the only error should be about a "bad rule" for clean-up commands.)

**Comment by ameshkov** [comment]
> gives the error

That's okay, there's nothing to delete yet.

**Comment by ameshkov** [comment]
> `/system/bin/ip6tables -t mangle -F PREROUTING`
> kills it.

What happens when you disable AdGuard? Do you need to reconnect to Wi-Fi to restore the Wi-Fi calling?

**Comment by carlylemiii** [comment]
> What happens when you disable AdGuard? Do you need to reconnect to Wi-Fi to restore the Wi-Fi calling?

The first couple of times it seemed that was the case, but now disabling AdGuard brings it back almost instantly.

> here goes one more build that's supposed to fix the issue

Unfortunately, no such luck.

**Comment by ameshkov** [comment]
> Unfortunately, no such luck.

Could you please collect AdGuard's default-level log with this new build?

**Comment by carlylemiii** [comment]
[adguard_logs_2.10.186_2201_0918.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1652457/adguard_logs_2.10.186_2201_0918.zip)

**Comment by ameshkov** [comment]
This must be smth new now.

Could you plz try the new set of commands? It'd be better to disable AG and make sure that Wi-Fi calling works before running the test.

### 0. Clean up

```
/system/bin/ip6tables -t mangle -D OUTPUT -j ADGUARD_OUTPUT
/system/bin/ip6tables -t mangle -D PREROUTING -j ADGUARD_PREROUTING
/system/bin/ip6tables -t mangle -F ADGUARD_OUTPUT
/system/bin/ip6tables -t mangle -F ADGUARD_PREROUTING
/system/bin/ip6tables -t mangle -X ADGUARD_OUTPUT
/system/bin/ip6tables -t mangle -X ADGUARD_PREROUTING
/system/bin/ip -6 rule del prio 8000 from all fwmark 26 lookup 800
/system/bin/ip -6 route flush table 800
```

### 1. Preparing the new rules table

```
/system/bin/ip6tables -t mangle -N ADGUARD_OUTPUT
/system/bin/ip6tables -t mangle -N ADGUARD_PREROUTING
/system/bin/ip6tables -t mangle -A OUTPUT -j ADGUARD_OUTPUT
/system/bin/ip6tables -t mangle -A PREROUTING -j ADGUARD_PREROUTING
```

### 2. Blocking QUIC
```
/system/bin/ip6tables -A OUTPUT -p udp --dport 80 -j DROP
/system/bin/ip6tables -A OUTPUT -p udp --dport 443 -j DROP
```

### 3. Adding new route
```
/system/bin/ip -6 rule add prio 8000 from all fwmark 26 lookup 800
/system/bin/ip -6 route add local default dev lo table 800
```

### 4. Protecting existing connections
```
/system/bin/ip6tables -t mangle -A ADGUARD_PREROUTING -i lo -p tcp -m socket -j RETURN
```

### 5. IPv6 redirect
These commands will redirect traffic to ports 80 and 443 to a non-existent proxy on port 1102. As there's no proxy on that port, it will effectively break IPv6, but anyway, it should have no effect on Wi-Fi calling in theory.

```
/system/bin/ip6tables -t mangle -A ADGUARD_OUTPUT -p tcp -d 2000::/3 --dport 80 -j MARK --set-mark 26
/system/bin/ip6tables -t mangle -A ADGUARD_PREROUTING -i lo -p tcp -d 2000::/3 --dport 80 -m mark --mark 26 -j TPROXY --on-port 1102 --on-ip ::1 --tproxy-mark 0x0
/system/bin/ip6tables -t mangle -A ADGUARD_OUTPUT -p tcp -d 2000::/3 --dport 443 -j MARK --set-mark 26
/system/bin/ip6tables -t mangle -A ADGUARD_PREROUTING -i lo -p tcp -d 2000::/3 --dport 443 -m mark --mark 26 -j TPROXY --on-port 1102 --on-ip ::1 --tproxy-mark 0x0
```

**Comment by carlylemiii** [comment]
Wi-Fi Calling remained functional with every single command...

**Comment by ameshkov** [comment]
Then the last build must've helped :( 

Could you please try it again, but this time reboot the device **after** installing it, just in case.

**Comment by carlylemiii** [comment]
I just tried rebooting and AdGuard still kicks me off Wi-Fi Calling :(

P.S. Things might be slightly better than they were before...from my past experience I always had to reboot to get Wi-Fi Calling functional again, at least now disabling AdGuard brings it back. Not that that solves the issue, but I thought I'd let you know.

**Comment by ameshkov** [comment]
This is a good sign, I feel that we're close.

There's only one difference between the commands AdGuard executes and the commands you've executed manually -- AdGuard redirects not just 80/443, but a lot more ports. I've added step 6 to the commands list, could you please check it?

**Comment by carlylemiii** [comment]
Do I need to do all of 1-5 again first?

**Comment by carlylemiii** [comment]
It took until the very end :)

`/system/bin/ip6tables -t mangle -A ADGUARD_PREROUTING -i lo -p tcp -d 2000::/3 --dport 8000:49151 -m mark --mark 26 -j TPROXY --on-port 1102 --on-ip ::1 --tproxy-mark 0x0
`

did it.

**Comment by carlylemiii** [comment]
So do it without adjusting port exclusions?

**Comment by carlylemiii** [comment]
[adguard_logs_2.10.186_2201_1200.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1653038/adguard_logs_2.10.186_2201_1200.zip)

**Comment by ameshkov** [comment]
> I actually just tried that a few minutes ago. That does keep it active.

Great, at least this one works :)

> What I wonder, and of course I'm spitballing here since I don't know the technical side of things: Is there a way to log all network activity in general or something, not just when AdGuard is active? I don't know if it matters that, since AdGuard proxy with those ports blocks Wi-Fi Calling, Wi-Fi Calling has to already be active before I actually enable proxy mode, so maybe AG is not catching Wi-Fi Calling actually becoming active?

In theory, we can use `tcpdump` to do it. The problem is that I doubt it will help us because I have not seen anything in the log file. This might mean that the mere presence of AdGuard rules breaks wi-fi calling.

We've got one more idea while reading the log file, one more test build's coming soon:)

**Comment by ameshkov** [comment]
@carlylemiii meanwhile, before the test build I've got one more idea.

In the last log file, I can see that you have some custom iptables rules redirecting port 5060 to some special tunnel.

Port 5060 is usually used for SIP protocol. Also, when SIP is in use, the ports range 10000-20000 is often used for audio transfer. I wonder will it help to exclude that ports range?

To check it, change the `pref.filtered.ports` value to this:
`80,443,3128,3130,3143,4443,5443,6443,7275,7443, 8000-9999,20001-49151`

**Comment by carlylemiii** [comment]
> I wonder will it help to exclude that ports range?

Just tried. Nope. :(

**Comment by ameshkov** [comment]
@carlylemiii 

<details>
<summary>Your ip6tables rules</summary>

```
Chain PREROUTING (policy ACCEPT), target     prot opt source               destination         , 
nm_mdmprxy_pkt_marker  all      ::/0                 ::/0                , 
nm_mdmprxy_icmp_pkt_marker  all      ::/0                 ::/0                , , 

Chain INPUT (policy ACCEPT), target     prot opt source               destination         , 
nm_mdmprxy_dl_ping6_marker  all      ::/0                 ::/0                , 
wakeupctrl_mangle_INPUT  all      ::/0                 ::/0                , 
routectrl_mangle_INPUT  all      ::/0                 ::/0                , , 

Chain FORWARD (policy ACCEPT), target     prot opt source               destination         , 
natctrl_mangle_FORWARD  all      ::/0                 ::/0                , , 

Chain OUTPUT (policy ACCEPT), target     prot opt source               destination         , , 

Chain POSTROUTING (policy ACCEPT), target     prot opt source               destination         , 
nm_mdmprxy_mngl_post  all      ::/0                 ::/0                , 
oem_mangle_post  all      ::/0                 ::/0                , 
bw_mangle_POSTROUTING  all      ::/0                 ::/0                , 
idletimer_mangle_POSTROUTING  all      ::/0                 ::/0                , 
qcom_qos_reset_POSTROUTING  all      ::/0                 ::/0                , 
qcom_qos_filter_POSTROUTING  all      ::/0                 ::/0                , , 

Chain bw_mangle_POSTROUTING (1 references), target     prot opt source               destination         
,            all      ::/0                 ::/0                 owner socket exists, , 

Chain idletimer_mangle_POSTROUTING (1 references), target     prot opt source               destination         
, IDLETIMER  all      ::/0                 ::/0                 timeout:10 label:0 send_nl_msg:1, 
IDLETIMER  all      ::/0                 ::/0                 timeout:15 label:1 send_nl_msg:1, , 

Chain natctrl_mangle_FORWARD (1 references), target     prot opt source               destination         , , 

Chain nm_mdmprxy_dl_ping6_marker (1 references), target     prot opt source               destination         , 
MARK       icmpv6    ::/0                 2600:1002:1120:ccc::/64  ipv6-icmptype 128 MARK set 0x7, 
ACCEPT     all      ::/0                 ::/0                 mark match 0x7, , 

Chain nm_mdmprxy_icmp_pkt_marker (1 references), target     prot opt source               destination         , 
CONNMARK   icmpv6    ::/0                 ::/0                 ctstate NEW CONNMARK set 0x9, , 

Chain nm_mdmprxy_mark_prov_chain (1 references), target     prot opt source               destination         , 
nm_mdmprxy_pkt_skmark  udp      ::/0                 ::/0                 socket --transparent --nowildcard --restore-skmark, 
nm_mdmprxy_pkt_skmark  tcp      ::/0                 ::/0                 socket --transparent --nowildcard --restore-skmark, , 

Chain nm_mdmprxy_mngl_post (1 references), target     prot opt source               destination         , 
TCPMSS     tcp      2600:1002:1120:ccc::/64  ::/0                 tcp flags:0x06/0x02 TCPMSS set 1140, 
MARK       all      ::/0                 ::/0                 mark match 0x9 MARK and 0x0, , 

Chain nm_mdmprxy_mngl_pre_ex (1 references), target     prot opt source               destination         , , 
Chain nm_mdmprxy_mngl_pre_spi (1 references), target     prot opt source               destination         , , 
Chain nm_mdmprxy_mngl_pre_tee (0 references), target     prot opt source               destination         , , 

Chain nm_mdmprxy_pkt_forwarder (1 references), target     prot opt source               destination         , 
TEE        all      ::/0                 2600:1002:1120:ccc::/64  TEE gw:fd95:d1e:12:: oif=r_rmnet_data1, 
DROP       all      ::/0                 2600:1002:1120:ccc::/64 , , 

Chain nm_mdmprxy_pkt_marker (1 references), target     prot opt source               destination         , 
CONNMARK   all      ::/0                 ::/0                 
CONNMARK restore, nm_mdmprxy_mark_prov_chain  all      ::/0                 ::/0                , 
nm_mdmprxy_mngl_pre_spi  all      ::/0                 ::/0                , 
nm_mdmprxy_mngl_pre_ex  all      ::/0                 ::/0                , 
MARK       tcp      ::/0                 ::/0                 tcp dpt:5060 MARK set 0x9, 
MARK       udp      ::/0                 ::/0                 udp dpt:5060 MARK set 0x9, 
CONNMARK   all      ::/0                 ::/0                 CONNMARK save, 
nm_mdmprxy_pkt_forwarder  all      ::/0                 ::/0                 mark match 0x9, , 

Chain nm_mdmprxy_pkt_skmark (2 references), target     prot opt source               destination         , 
RETURN     all      ::/0                 ::/0                , , 

Chain oem_mangle_post (1 references), target     prot opt source               destination         , , 
Chain qcom_qos_filter_POSTROUTING (1 references), target     prot opt source               destination         , , 
Chain qcom_qos_reset_POSTROUTING (1 references), target     prot opt source               destination         , 
MARK       all      2600:100a:103a:64d0:192c:e844:ea12:3a68  ::/0                 MARK and 0x0, 
MARK       all      2600:100a:1127:f9ca:cca7:2f5f:a14b:5f65  ::/0                 MARK and 0x0, 
MARK       all      2600:1002:1120:ccc:3ed2:c6df:7f98:a7e7  ::/0                 MARK and 0x0, , 

Chain routectrl_mangle_INPUT (1 references), target     prot opt source               destination         , 
MARK       all      ::/0                 ::/0                 MARK set 0x3000b, 
MARK       all      ::/0                 ::/0                 MARK set 0x30075, 
MARK       all      ::/0                 ::/0                 MARK set 0x3000a, 
MARK       all      ::/0                 ::/0                 MARK set 0x3000a, 
MARK       all      ::/0                 ::/0                 MARK set 0xf0078, , 

Chain wakeupctrl_mangle_INPUT (1 references), target     prot opt source               destination         , 
NFLOG      all      ::/0                 ::/0                 mark match 0x80000000/0x80000000 limit: avg 10/sec burst 5 nflog-prefix  "iface:wlan0" nflog-group 3 nflog-threshold 8
```

</details>

<br/>
Intensive googling shows that these rules come from a proprietary Qualcomm tool netmgrd.

<br/><br/>
<details>
<summary>Useful technical details (saving it for the future)</summary>

Here's how it sets up iptables rules:
https://github.com/redheavenliu/data/blob/master/netmgr/src/plugins/modem_proxy/nm_mdmprxy_legacy_mark_provider.c

Here are all the constants it uses:
https://github.com/redheavenliu/data/blob/master/netmgr/src/plugins/modem_proxy/nm_mdmprxy.h

```
#define NM_MDMPRXY_MODEM_PORT_START       32000
#define NM_MDMPRXY_MODEM_PORT_END         36999

/* IMS port-range used for tcp handhsake */
#define NM_MDMPRXY_IMS_PORT_START         32100
#define NM_MDMPRXY_IMS_PORT_END           32600

/* IMS uses a random port between 32100 to 32600 for doing TCP handshake
   We have an iptables rule to mark and forward packets within this range
   The port uses for IKE and ESP needs to be outside this range to prevent
   collision */
#define NM_MDMPRXY_MODEM_PORT_RANGE_START 32601
#define NM_MDMPRXY_MODEM_PORT_RANGE_END   36999
```
</details>

<br/><br/>
Let's try excluding the range 32000-36999 now:
`80,443,3128,3130,3143,4443,5443,6443,7275,7443,8000-31999,37000-49151`

**Comment by carlylemiii** [comment]
You're really digging into this, and I appreciate that! That exclusion didn't work, though :(

I just noticed if I have an active call going and then enable AG, the call doesn't drop. Would an AG log in that case offer anything useful, or no because the call's already been started prior?

**Comment by ameshkov** [comment]
> I just noticed if I have an active call going and then enable AG, the call doesn't drop. Would an AG log in that case offer anything useful, or no because the call's already been started prior?

No, the connection is already active, AG won't mess with it.

**Comment by ameshkov** [comment]
We're on the right way, but we need to conduct a few more experiments and we rly need a new build for that:)

**Comment by ameshkov** [comment]
Thank you very much for helping us!

**Comment by carlylemiii** [comment]
My pleasure!

**Comment by ameshkov** [comment]
Good morning @carlylemiii !:)

Finally, we've prepared one more test build:
[adguard_ipv6_fix4.apk.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1657465/adguard_ipv6_fix4.apk.zip)

Some tech details. My mistake was that I thought `32000-36999` is a range of the remote server ports. In reality, this is a range of local ports, and it can't be excluded that easily. Also, we shouldn't exclude these ports on every device. Instead, we should check if `netmgrd` is present and these local ports are restricted to regular TCP traffic.

**Comment by carlylemiii** [comment]
No luck :(

One thing I noticed is occasionally, after disabling AdGuard and re-enabling, I can make one successful call, and then a subsequent call fails, as usual. I bet this is just some sort or race condition, that I'm beating AdGuard to being fully enabled, but I thought I'd let you know anyway.

**Comment by carlylemiii** [comment]
Let me try a reboot, just in case :)

**Comment by ameshkov** [comment]
Oh, I forgot to ask you to record the debug-level log with this build.

**Comment by carlylemiii** [comment]
Sure thing.

[adguard_logs_2.10.186_2301_1451.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1657579/adguard_logs_2.10.186_2301_1451.zip)

**Comment by ameshkov** [comment]
Hmm, in your case this ports range was not forbidden, and that's why our solution didn't help -- it simply was not used:(

**Comment by ameshkov** [comment]
Ok, one more try:)

This time we used a very simple solution -- Verizon's IPv6 range supposedly used for Wi-Fi calling is excluded (`2600:1000::/28`, `2600:1000::/29`).

[adguard_ipv6_fix5.apk.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/1657749/adguard_ipv6_fix5.apk.zip)

**Comment by carlylemiii** [comment]
Trying it now. Do you want a log?

**Comment by ameshkov** [comment]
Only if it does not work:)

**Comment by carlylemiii** [comment]
Fifth time's the charm... No unintended consequences with that expected? :D

**Comment by ameshkov** [comment]
Yay, finally, thank you!

We should now expose the hardcoded range via a low-level setting flag. The main issue with that solution is that we need ppl help to figure what exact IP ranges should be excluded for other providers.

> No unintended consequences with that expected

Not really.

**Comment by sfionov** [comment]
Fixed in 2.11beta
